meeteval/wer/api.py

```diff
@@ -0,0 +1,179 @@
+import glob
+import logging
+import re
+from pathlib import Path
+
+import meeteval.io
+from meeteval.wer.wer import ErrorRate
+
+__all__ = [
+    'cpwer',
+    'orcwer',
+    'mimower',
+    'tcpwer',
+]
+
+
+def _load_texts(
+        reference_paths: 'list[str]',
+        hypothesis_paths: 'list[str]',
+        regex,
+        reference_sort: 'bool | str' = False,
+        hypothesis_sort: 'bool | str' = False,
+        file_format=None,
+) -> 'tuple[meeteval.io.SegLST, meeteval.io.SegLST]':
+    """Load and validate reference and hypothesis texts.
+
+    Validation checks that reference and hypothesis have the same example IDs.
+    """
+
+    # Normalize and glob (for backwards compatibility) the path input
+    def _glob(pathname):
+        match = list(glob.glob(pathname))
+        # Forward pathname if not matched to get the correct error message
+        return match or [pathname]
+
+    def load(paths) -> meeteval.io.SegLST:
+        if isinstance(paths, (str, Path)):
+            paths = [paths]
+        if isinstance(paths, (tuple, list)):
+            paths = [Path(file) for r in paths for file in _glob(str(r))]
+            return meeteval.io.asseglst(
+                meeteval.io.load(paths, format=file_format))
+        elif isinstance(paths, meeteval.io.base.BaseABC):
```

Comment:
I'm not sure whether this is the correct check. There is a todo on my list for a converter for dataframes, which don't inherit `meeteval.io.base.BaseABC`

Reply:
I will change it to try `meeteval.io.asseglst` and raise an TypeError, if that function raises an Excpetion.